nix/cabal2nix.nix

```diff
@@ -10,21 +10,22 @@
 #
 { pkgs, cabal, flags }:
 with pkgs.lib;
-
 let
   # The package name derived from the cabal file name:
   name = removeSuffix ".cabal" (baseNameOf (toString cabal));
 
   # All flags as a string:
   flagsStr = concatMapStringsSep " " (f: "-f${f}") flags;
 
-in pkgs.stdenvNoCC.mkDerivation {
-  name = "${name}.nix";
-  src = cabal;
-
+in
+pkgs.stdenvNoCC.mkDerivation {
+  name = "${name}-cabal2nix";
+  phases = [ "buildPhase" ];
   buildInputs = with pkgs; [ cabal2nix ];
 
   buildCommand = ''
-    cabal2nix ${flagsStr} $(dirname ${cabal}) > $out
+    mkdir -p $out
+    ln -s ${cabal} $out/${name}.cabal
```

Comment:
With this solution, a link to the cabal file will be included in the the output. Which I guess doesn't hurt, but wouldn't it make more sense to put it in the temporary build directory?

```
ln -s ${cabal} > ./${name}.cabal
cabal2nix ${flagsStr} . > $out/default.nix
```

On the other hand, i noticed that the `src` attribute of the generated nix file refers to the out dir, so maybe it needs to have a reference to it? My version above seems to put `src = ./.` in the generated nix file, which I am not sure if it is better or worse.

The question is why it works at all, when the `src` attribute refers to a location that doesn't have the source?

Reply:
The `src` attribute in the generated Nix file is overridden by nix-hs to point back at the original source directory after passing through a filter to remove non-source files.  (You can see this in `nix/lib.nix` in the `cleanSource` function.)  So the final derivation will have the correct `src` attribute.  This is important to avoid rebuilds if a non-source file changes.

As far as where to link the cabal file, since the file is already put in the Nix store I'm not sure it matters.  That said, it's a simple change if needed.

Reply:
Ah, that makes sense. I see. Thanks!